website/server/controllers/top-level/payments/amazon.js

```diff
@@ -231,7 +235,18 @@ api.subscribeCancel = {
   middlewares: [authWithUrl],
   async handler (req, res) {
     let user = res.locals.user;
+    let groupId = req.query.groupId;
+
     let billingAgreementId = user.purchased.plan.customerId;
+    let planId = user.purchased.plan.planId;
+    let lastBillingDate = user.purchased.plan.lastBillingDate;
+    if (groupId) {
+      let group = await Group.findById(groupId).exec();
```

Comment:
limit returned fields? at least remove the chat. Also there's no handling if the group isn't found or on who's cancelling the subscription to have the necessary permissions